pyiceberg/table/inspect.py

```diff
@@ -657,3 +669,35 @@ def all_manifests(self) -> "pa.Table":
             lambda args: self._generate_manifests_table(*args), [(snapshot, True) for snapshot in snapshots]
         )
         return pa.concat_tables(manifests_by_snapshots)
+
+    def _all_files(self, data_file_filter: Optional[Set[DataFileContent]] = None) -> "pa.Table":
+        import pyarrow as pa
+
+        snapshots = self.tbl.snapshots()
+        if not snapshots:
+            return pa.Table.from_pylist([], schema=self._get_files_schema())
+
+        executor = ExecutorFactory.get_or_create()
+        all_manifest_files_by_snapshot: Iterator[List[ManifestFile]] = executor.map(
+            lambda args: args[0].manifests(self.tbl.io), [(snapshot,) for snapshot in snapshots]
+        )
+        all_manifest_files = list(
+            {(manifest.manifest_path, manifest) for manifest_list in all_manifest_files_by_snapshot for manifest in manifest_list}
+        )
+        all_files_by_manifest: Iterator[List[Dict[str, Any]]] = executor.map(
+            lambda args: self._files_by_manifest(*args), [(manifest, data_file_filter) for _, manifest in all_manifest_files]
+        )
+        all_files_list = [file for files in all_files_by_manifest for file in files]
+        return pa.Table.from_pylist(
+            all_files_list,
+            schema=self._get_files_schema(),
+        )
```

Comment:
WDYT about something like this?

Also i would rename `_files_by_manifest` and have it return pa.Table, so we can skip the flatten and just concat the tables.


```suggestion
        manifest_lists = executor.map(
            lambda snapshot: snapshot.manifests(self.tbl.io),
            snapshots
        )
        
        unique_manifests = {
            (manifest.manifest_path, manifest)
            for manifest_list in manifest_lists
            for manifest in manifest_list
        }
        
        file_lists = executor.map(
            self._files_by_manifest,
            [(manifest, data_file_filter) for _, manifest in unique_manifests]
        )
        
        all_files = [
            file 
            for file_list in file_lists 
            for file in file_list
        ]
        
        return pa.Table.from_pylist(
            all_files,
            schema=self._get_files_schema()
        )
```

Reply:
I agree with this, the impl of the `_files_by_manifest` enforces uniqueness which wasn't clear